snap/hooks/configure

```diff
@@ -371,4 +372,39 @@ then
     then
         chgrp microk8s -R "$SNAP_DATA/juju" "$SNAP_DATA/juju-home" || true
     fi
-fi
\ No newline at end of file
+fi
+
+# Set up a bridge with a known IP address that services can bind to instead of
+# whatever IP gets handed out by their router. Doesn't affect services that bind
+# to 127.0.0.1.
+API_IP="$(snapctl get api.ip)"
+if [[ "$API_IP" == "" ]]
+then
+  API_IP="10.5.2.1"
+  snapctl set api.ip="${API_IP}"
+fi
+
+if ADDR_INFO="$(sudo ip -j a show uk8s0 2>/dev/null)"; then
+  CURRENT_IP="$(echo $ADDR_INFO | jq -r '.[0].addr_info[] | select(.family == "inet").local')"
+
+  if [[ "$API_IP" != "$CURRENT_IP" ]]; then
+    $SNAP/sbin/ip link delete uk8s0
+    $SNAP/sbin/ip link add uk8s0 type bridge
+    $SNAP/sbin/ip address add $API_IP/24 dev uk8s0
+    $SNAP/sbin/ip link set uk8s0 up
+
+    for conf in client controller kubelet proxy scheduler; do
+      $SNAP/bin/sed -i 's/IP/'"${API_IP}"'/g' ${SNAP_DATA}/credentials/${conf}.config
+      $SNAP/bin/sed -i 's|server: .*|server: https://'"${API_IP}"':16443|g' ${SNAP_DATA}/credentials/${conf}.config
```

Comment:
This does not look right in the case of a cluster node. Lets say this microk8s instance is a node part of a cluster and the apiserver is a remote server. We will be telling kubelet to contact the local apiserver that is not running.

Reply:
@ktsakalozos: How do we figure out if microk8s is clustered, and how do we figure out what the correct IP is?

Reply:
If a node act like a cluster it has the ${SNAP_DATA}/var/lock/clustered.lock file, see https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/wrappers/run-etcd-with-args#L10